RELEASE/scripts/autoscend/iotms/mr2021.ash

```diff
@@ -112,3 +112,245 @@ int auto_backupUsesLeft()
+		{
+			craft("combine", 1, $item[battery (lantern)], $item[battery (AAA)]);
+			return (available_amount($item[battery (car)]) >= 1);
+		}
+		// Single step.
+		else if (available_amount($item[battery (9-Volt)]) >= 1 && available_amount($item[battery (AA)]) >= 1)
+		{
+			craft("combine", 1, $item[battery (9-Volt)], $item[battery (AA)]);
+			return (available_amount($item[battery (car)]) >= 1);
+		}
+		// Single step.
+		else if (available_amount($item[battery (D)]) >= 2)
+		{
+			craft("combine", 1, $item[battery (D)], $item[battery (D)]);
+			return (available_amount($item[battery (car)]) >= 1);
+		}
+		// The only multi-step case that can't be resolved by the same function (can't turn AAs into a lantern without a AA or D)
+		else if (available_amount($item[battery (AA)]) >= 3)
+		{
+			batteryCombine($item[battery (9-volt)]);
+			craft("combine", 1, $item[battery (9-Volt)], $item[battery (AA)]);
+			return (available_amount($item[battery (car)]) >= 1);
+		}
+		// Every other multi step case with recusion.
+		else if (available_amount($item[battery (AAA)]) >= 6 ||
+		 (available_amount($item[battery (AA)]) >= 1 && available_amount($item[battery (AAA)]) >= 4) ||
+		 (available_amount($item[battery (D)]) >= 1 && available_amount($item[battery (AAA)]) >= 3) ||
+		 (available_amount($item[battery (9-Volt)]) >= 1 && available_amount($item[battery (AAA)]) >= 2) ||
+		 (available_amount($item[battery (AA)]) >= 2 && available_amount($item[battery (AAA)]) >= 2) ||
+		 (available_amount($item[battery (D)]) >= 1 && available_amount($item[battery (AA)]) >= 1 && available_amount($item[battery (AAA)]) >= 1))
+		{
+			batteryCombine($item[battery (lantern)]);
+			craft("combine", 1, $item[battery (lantern)], $item[battery (AAA)]);
+			return (available_amount($item[battery (car)]) >= 1);
+		}
+	}
+	return false;
+}
+
+// This function will ensure a battery is available before use, if possible.
+boolean auto_getBattery(item battery)
+{
+	// If we have the desired battery then we're done here.
+	if (available_amount(battery) >= 1)
+	{
+		return true;
+	}
+		
+	// We'll try to create the battery, if it works then great, if not, we keep going.
+	if (batteryCombine(battery))
+	{
+		return true;
+	}
+
+	// We need to check that we can untinker (handled by totalBatteryPoints), 
+	// then make sure we have enough AAA value to craft the target battery.
+	if (totalBatteryPoints() > batteryPoints(battery))
+	{
+		// We're going to break down each battery one at a time from largest to smallest, and attempt to craft.
+		foreach it in $items[battery (car), battery (lantern), battery (9-Volt), battery (D), battery (AA)]
+		{
+			for i from 1 to available_amount(it)
+			{
+				visit_url("place.php?whichplace=forestvillage&action=fv_untinker&pwd=&preaction=untinker&whichitem=" + it.to_int().to_string());
+				
+				if (batteryCombine(battery))
+				{
+					return true;
+				}
+			}
+		}
+	}
+	return false;

```

Comment:
probably want to make sure untinkering is available before we try this and I'm not sure that if should be in the middle of the for loop there as I think there are scenarios it can get stuck in infinite loops with repeatedly half crafting what is asked for then untinkering  (I may be wrong here haven't stepped though all the logic).
Maybe make it just untinker everything down to AAA's first (if untinkering is available) and then make the requested battery as we know that will always work. I guess it might waste a few tens of meat on paste to remake something we untinkered but if that's an issue the user likely has bigger problems.